PhotoKeeper/PhotoGridViewController.swift

```diff
@@ -7,114 +7,118 @@
         navBar?.titleTextAttributes =  [NSAttributedStringKey.foregroundColor: black]
     }
-    
+
     @objc func showCameraViewController(){
         let cvc = CameraViewController()
         let nvc = UINavigationController(rootViewController: cvc)
         present(nvc, animated: true, completion: nil)
     }
 
-    override func didReceiveMemoryWarning() {
-        super.didReceiveMemoryWarning()
-        // Dispose of any resources that can be recreated.
-    }
-    
     // MARK: UICollectionViewDataSource
     override func collectionView(_ collectionView: UICollectionView,
                                  didSelectItemAt indexPath: IndexPath) {
-        let item = images[indexPath.row]
+        let photo = photos[indexPath.row]
         let pvc = PhotoViewController()
-        let imageName = images[indexPath.row].fileName
-        let imagePath = documentsDirectory.appendingPathComponent("\(imageName)")
-        pvc.image = UIImage(contentsOfFile: imagePath.path)
-        pvc.imageTitle = item.title
-        pvc.imageDescription = item.description
+        pvc.photo = photo
         navigationController?.pushViewController(pvc, animated: true)
     }
-    
+
     override func numberOfSections(in collectionView: UICollectionView) -> Int {
         return 1
     }
-    
+
     override func collectionView(_ collectionView: UICollectionView,
                                  numberOfItemsInSection section: Int) -> Int {
-        return images.count
+        return photos.count
     }
 
     override func collectionView(_ collectionView: UICollectionView,
                                  cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
         let cell = collectionView.dequeueReusableCell(withReuseIdentifier: reuseIdentifier, for: indexPath)
-        let imageName = images[indexPath.row].fileName
-        let imagePath = documentsDirectory.appendingPathComponent("\(imageName)")
-        if FileManager.default.fileExists(atPath: imagePath.path) {
-            let image = UIImage(contentsOfFile: imagePath.path)
-            let iv = UIImageView(image: image)
-            iv.contentMode = .scaleAspectFill
+        let photo = photos[indexPath.row]
+        let photoPath = store.getPathForPhoto(photo)
+        if FileManager.default.fileExists(atPath: photoPath) {
+            let image = UIImage(contentsOfFile: photoPath)
+            let imageView = UIImageView(image: image)
+            imageView.contentMode = .scaleAspectFill
             cell.clipsToBounds = true
-            iv.frame = CGRect(x: 0, y: 0, width: cell.frame.width, height: cell.frame.height)
-            cell.backgroundView = iv
+            imageView.frame = CGRect(x: 0, y: 0, width: cell.frame.width, height: cell.frame.height)
+            cell.backgroundView = imageView
+        }
+
+        /* hacky remove all the title content things idfk..... */
+
+        for view in cell.contentView.subviews {
+            view.removeFromSuperview()
```

Comment:
this works for now. but we'll want to move layout and logic to a custom cell. that will prevent hacks like this

Reply:
hey! yeah i tried to create a custom cell but was running into the same problem where it was getting initalized and never updating; properly and would just end up on random grid cells... would love to have a chat about custom cells / data reloading / cell lifecycles !!!